syntax = "proto3";

package image_analyzer;

option go_package = "destinyface/gen/proto";

service ImageAnalyzer {
    rpc AnalyzeImage (AnalyzeRequest) returns (AnalyzeResponse);
}

message AnalyzeRequest {
    bytes image_data = 1;
}

message AnalyzeResponse {
    string product_name = 1;
    repeated float vector = 2;
    bool success = 3;
}